### PR TITLE
Support emulation of Firestore functions

### PR DESCRIPTION
**Pull request by laurenzlong**
### Description

This PR supports emulation of Firestore functions.

### Sample Commands

```
firebase experimental:functions:shell
```

Inside shell:

```
# invoke onCreate function
myFirestoreFunction({foo: ‘new’})

# invoke onDelete function
myFirestoreFunction({foo: ‘old’})

# invoke onUpdate or onWrite function
myFirestoreFunction({before: {foo: ‘old’}, after: {foo: ‘new’} })

# mock wildcards in path, for example: if the path was input/{group}/{id}
myFirestoreFunction({foo: ‘new’}, {params: {group: 'a', id: 123}})
```

### CODE REVIEW

**Comment by merlinnot** [comment]
@laurenzlong so happy to see this PR ❤️

**Comment by laurenzlong** [comment]
Comments addressed.

File: lib/localFunction.js [hunk 1]
```diff
@@ -48,140 +46,65 @@ LocalFunction.prototype._requestCallBack = function(err, response, body) {
...
-            fields: this._encodeFirestoreValue(data.after),
-            createTime: currentTime,
-            updateTime: currentTime
-          },
-          oldValue: {
-            fields: this._encodeFirestoreValue(data.before),
-            createTime: currentTime,
-            updateTime: currentTime
-          }
+          value: {},
```
Comment (mbleigh) [inline lib/localFunction.js]: Just checking: the correct value here is an empty object, not `null`?

Comment (laurenzlong) [reply lib/localFunction.js]: yep

**Comment by renearias** [comment]
data exist on write event and before null

**Comment by rjhuijsman** [comment]
Hi @renearias - are you reporting a problem? If so, can you please file a GitHub issue with more detail?

**Comment by renearias** [comment]
ok @rjhuijsman  thanks

**Pull request merged**